napari/utils/tree/group.py

```diff
@@ -38,9 +40,35 @@ def __init__(
     ):
         Node.__init__(self, name=name)
         SelectableNestableEventedList.__init__(
-            self, data=children, basetype=basetype
+            self,
+            data=children,
+            basetype=basetype,
+            lookup={str: lambda e: e.name},
         )
 
+    def __newlike__(self, iterable: Iterable):
+        # NOTE: TRICKY!
+        # whenever we slice into a group with group[start:end],
+        # the super().__newlike__() call is going to create a new object
+        # of the same type (Group), and then populate it with items in iterable
+        # ...
+        # However, `Group.insert` changes the parent of each item as
+        # it gets inserted.  (The implication is that no Node can live in
```

Comment:
> The implication is that no Node can live in multiple groups at the same time

Are we sure we want this? Do I understand right that this prevents a group from having multiple parents? Might that be something we want for sharing data across canvases/ across viewers?

Reply:
From discussion in community meeting - sounds like we don't need this here in the tree itself, we could think about other ways to solve this problem like datasources - right @tlambert03?

Reply:
no, the point i was making in the discussion is the we _do_ need this here:  i.e. in a true tree structure, (which Group/Node are modeling soas to work with a true Tree view like QTree view), then each node cannot have more than one parent.  Remember that this object isn't _just_ for layers, it's for any nested list, many of which are true tree structures.  When we use it to implement a LayerTree, we can at that point decide if the "layer only has one parent" restriction is too strict.  I _hope_ that won't be our case, because that would add a significant degree of complexity.  But ways to deal with that include:
- no longer a 1-to-1 relationship between `data` and `Layer`.  With `datasource`, each `Layer` can still have one-and-only-one-parent, but 2 layers can share the same data
- Use composition instead of inheritance: `LayerList` itself wouldn't actually be a `Group`, and `Layer` wouldn't actually be a `Node`.  But we'd add some intermediate `Node` object that _has_ a `Layer`.  Then two independent `Nodes` could have the same `Layer`
- last (and hopefuly not this one), we decide that the layerlist really isn't a tree.  it's a possibly cyclic graph... in which case I'm not sure we can use a list-like API at all anymore :joy:  it's

Reply:
oh sorry - yeah, when I said "don't need" i meant "don't need ability to support multiple parents in trees" but in context i guess it came across as "don't need the current behavior". Current behavior is fine, for reasons you outlined

Reply:
a reminder why one shouldn't write an unqualified "this" in their sentences 🤣